internal/tools/argocd/argocd_test.go

```diff
@@ -0,0 +1,128 @@
+		ClusterTypes: "customer",
+		ClusterID:    "1234567",
+	}
+
+	suite.argoK8sFile = &Argock8sRegister{
+		Clusters: []ArgocdClusterRegisterParameters{
+			{
+				Name:      "cluster1",
+				Type:      "kops",
+				Labels:    newClusterLabels,
+				APIServer: "cluster1.test.com",
+				CaData:    base64.StdEncoding.EncodeToString([]byte("-----BEGIN PRIVATE KEY-----MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCTUCgR5+EsitNh-----END PRIVATE KEY-----")),
+				CertData:  base64.StdEncoding.EncodeToString([]byte("-----BEGIN CERTIFICATE-----MIIDGTCCAgGgAwIBAgIUHqQQpkxCJ/xg6G/PVyFFEYrBPjswDQYJKoZIhvcNAQEL-----END CERTIFICATE-----")),
+				KeyData:   base64.StdEncoding.EncodeToString([]byte("-----BEGIN PRIVATE KEY-----MIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQDNPAqO0X1O7gw-----END PRIVATE KEY-----")),
+			},
+		},
+	}
+	suite.newArgoK8sFile = &ArgocdClusterRegisterParameters{
+		Name:      "cluster2",
+		Type:      "kops",
+		Labels:    newClusterLabels,
+		APIServer: "cluster2.test.com",
+		CaData:    base64.StdEncoding.EncodeToString([]byte("-----BEGIN PRIVATE KEY-----MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCTUCgR5+EsitNh-----END PRIVATE KEY-----")),
+		CertData:  base64.StdEncoding.EncodeToString([]byte("-----BEGIN CERTIFICATE-----MIIDGTCCAgGgAwIBAgIUHqQQpkxCJ/xg6G/PVyFFEYrBPjswDQYJKoZIhvcNAQEL-----END CERTIFICATE-----")),
+		KeyData:   base64.StdEncoding.EncodeToString([]byte("-----BEGIN PRIVATE KEY-----MIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQDNPAqO0X1O7gw-----END PRIVATE KEY-----")),
+	}
+
+	suite.updatedArgok8sFile = &Argock8sRegister{
+		Clusters: []ArgocdClusterRegisterParameters{
+			{
+				Name:      "cluster1",
+				Type:      "kops",
+				Labels:    newClusterLabels,
+				APIServer: "cluster1.test.com",
+				CaData:    base64.StdEncoding.EncodeToString([]byte("-----BEGIN PRIVATE KEY-----MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCTUCgR5+EsitNh-----END PRIVATE KEY-----")),
+				CertData:  base64.StdEncoding.EncodeToString([]byte("-----BEGIN CERTIFICATE-----MIIDGTCCAgGgAwIBAgIUHqQQpkxCJ/xg6G/PVyFFEYrBPjswDQYJKoZIhvcNAQEL-----END CERTIFICATE-----")),
+				KeyData:   base64.StdEncoding.EncodeToString([]byte("-----BEGIN PRIVATE KEY-----MIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQDNPAqO0X1O7gw-----END PRIVATE KEY-----")),
+			},
+			{
+				Name:      "cluster2",
+				Type:      "kops",
+				Labels:    newClusterLabels,
+				APIServer: "cluster2.test.com",
+				CaData:    base64.StdEncoding.EncodeToString([]byte("-----BEGIN PRIVATE KEY-----MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCTUCgR5+EsitNh-----END PRIVATE KEY-----")),
+				CertData:  base64.StdEncoding.EncodeToString([]byte("-----BEGIN CERTIFICATE-----MIIDGTCCAgGgAwIBAgIUHqQQpkxCJ/xg6G/PVyFFEYrBPjswDQYJKoZIhvcNAQEL-----END CERTIFICATE-----")),
+				KeyData:   base64.StdEncoding.EncodeToString([]byte("-----BEGIN PRIVATE KEY-----MIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQDNPAqO0X1O7gw-----END PRIVATE KEY-----")),
+			},
+		},
+	}
```

Comment:
Should we be reading these keys/certs in from somewhere like an env variable or config flag? 

If not, a small optimization might be to create a variable for `caData` `certData` and `keyData`, that way the base64 encoding only needs to happen once, and everything might be a bit more readble (and easier to change in the future if necessary)

Reply:
Changed, thanks @nickmisasi